### PR TITLE
add modbus mapper

### PR DESCRIPTION
**Pull request by RicardoZPHuang**
**What type of PR is this?**
> /kind feature

**What this PR does / why we need it**:
Add device mapper for modbus protocol devices

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: device/modbus_mapper/README.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+#    2. Replace <your_dockerhub_username> with your dockerhub username at spec.template.spec.containers.image
+
+kubectl create -f deployment.yaml
+```
+
+## Modules
+
+The modbus mapper consists of the following four major modules :-
+
+ 1. Controller
```
Comment (rohitsardesai83) [inline device/modbus_mapper/README.md]: We have the mapper design : https://github.com/kubeedge/kubeedge/blob/master/docs/proposals/mapper-design.md

And a bluetooth protocol mappers implemented based on this design : 

https://github.com/kubeedge/kubeedge/tree/master/device/bluetooth_mapper

Is it possible that we can model the modbus mappers based on the mapper design architecture ?

File: device/modbus_mapper/README.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+ File Watcher is a component which can load dpl and mqtt configuration from configuration files.The following are the main responsibilities of this component: 
+ a) To monitor the dpl configuration file. If this file changed, file watcher will reload the dpl configuration to the mapper.
+
+ b) To load dpl and mqtt configuration when mapper starts first time.
+
+ ## Usage
+ 
+ ### Configuration File
+ 
+ The user can give the configurations specific to the modbus device using dpl configuration provided in the dpl file present at $GOPATH/src/github.com/kubeedge/kubeedge/device/mosbua_mapper/dpl/dpl.
```
Comment (rohitsardesai83) [inline device/modbus_mapper/README.md]: mosbua_mapper  -> modbus_mapper ?

File: device/modbus_mapper/README.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+ The details provided in the dpl configuration file are used by devicetwin-manager module, modbus-manager module, file watcher module, and the controller.
+ 
+ **Example:** Given below is the instructions using which user can create their own dpl configuration file, for their device.
```
Comment (rohitsardesai83) [inline device/modbus_mapper/README.md]: Given below ~is~ are the instructions

File: device/modbus_mapper/README.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+### Deploying modbus mapper application
+
+```shell
+cd $GOPATH/src/github.com/kubeedge/kubeedge/device/modbus_mapper
+
+# Please enter the following details in the modbus-dpl-configmap.yaml :-
+#    1. Replace <edge_node_name> with the name of your edge node at meta.name
+#    2. Replace <dpl-configuration> with your device protocol configuration at data.dpl
+
+kubectl create -f modbus-dpl-configmap.yaml
```
Comment (rohitsardesai83) [inline device/modbus_mapper/README.md]: Does the mapper not use the config map cretated by the device controller in the cloud ? if we create the device model and device instance for the modbus device , then the device controller in the cloud can create the required config map and push it to the edge node.

File: device/modbus_mapper/README.md (hunk 2 above)
Comment (RicardoZPHuang) [reply device/modbus_mapper/README.md]: typo fixed

File: device/modbus_mapper/README.md (hunk 4 above)
Comment (RicardoZPHuang) [reply device/modbus_mapper/README.md]: change to using deviceProfile.json created by device controller.

File: device/modbus_mapper/README.md (hunk 3 above)
Comment (RicardoZPHuang) [reply device/modbus_mapper/README.md]: fixed

File: device/modbus_mapper/README.md (hunk 1 above)
Comment (RicardoZPHuang) [reply device/modbus_mapper/README.md]: Modbus mapper is developed by nodejs. It is a little hard to model in that way. I checked with Kevin it's ok if it works with device-crd. Maybe reconstruction by golang later can model according to the above architecture.

Comment (rohitsardesai83) [reply device/modbus_mapper/README.md]: ok

File: device/modbus_mapper/src/devicetwin.js [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+                break;
+            default:
+                logger.error('unknown dataType: ', property.datatype);
+                callback(null);
+                break;    
+        }
+    }
+
+    // updateActual update actual value to edge mqtt
+    updateActual(deviceID, property, value) {
```
Comment (rohitsardesai83) [inline device/modbus_mapper/src/devicetwin.js]: just fyi , we have changed from "expected/actual" to "desired/reported" in CRDs.

Comment (rohitsardesai83) [reply device/modbus_mapper/src/devicetwin.js]: this is still not reflected on edge side , overall looks inconsistent from UX.

Comment (RicardoZPHuang) [reply device/modbus_mapper/src/devicetwin.js]: The edge side devicetwin is still using the "expected/actual", so just keep this format

**Pull request merged**